tests/behaviour/mod.rs

```diff
@@ -82,11 +92,39 @@ impl Context {
         t == "ignore" || t == "ignore-typedb" || t == "ignore-client-rust" || t == "ignore-typedb-client-rust"
     }
 
-    async fn after_scenario(&self) -> TypeDBResult {
-        try_join_all(self.databases.all().await.unwrap().into_iter().map(Database::delete)).await?;
+    pub async fn after_scenario(&mut self) -> TypeDBResult {
+        sleep(Context::PAUSE_BETWEEN_STEP_CHECKS).await;
+        self.set_connection(Connection::new_encrypted(
+            &["localhost:11729", "localhost:21729", "localhost:31729"],
+            Credential::with_tls(&Context::ADMIN_USERNAME, &Context::ADMIN_PASSWORD, Some(&self.tls_root_ca))?,
+        )?);
+        self.cleanup_databases().await;
+        self.cleanup_users().await;
         Ok(())
     }
 
+    pub async fn all_databases(&self) -> HashSet<String> {
+        self.databases.all().await.unwrap().into_iter().map(|db| db.name().to_owned()).collect::<HashSet<_>>()
+    }
+
+    pub async fn cleanup_databases(&mut self) {
+        try_join_all(self.databases.all().await.unwrap().into_iter().map(Database::delete)).await.ok();
+    }
```

Comment:
We ignore possible errors that could appear when database is still in `all()` but is already deleted.